lib/mix/lib/mix/tasks/escript.install.ex

```diff
@@ -0,0 +1,113 @@
+defmodule Mix.Tasks.Escript.Install do
+  use Mix.Task
+
+  @shortdoc "Install an escript locally"
+
+  @moduledoc """
+  Install an escript locally.
+
+  If no argument is supplied but there is an escript in the project's root directory
+  (created with `mix escript.build`), then the escript will be installed
+  locally. For example:
+
+      mix do escript.build, escript.install
+
+  The argument is usually a package name and optionally a version:
+
+      mix escript.install phoenix
+      mix escript.install phoenix 0.1.2
+
+  It could also be a remote repository:
+
+      mix escript.install git://example.com/foo.git          # infers Git from the URL
+      mix escript.install https://example.com/foo.git        # infers Git from the URL
+      mix escript.install github.com/example/foo             # fetches git://github.com/example/example.git
+      mix escript.install https://github.com/example/foo     # fetches https://github.com/example/example.git
+      mix escript.install --scm=git example.com/foo          # fetches git://example.com/example
+      mix escript.install --scm=git https://example.com/foo  # fetches https://example.com/example
```

Comment:
> I like @ericmj's proposal here. With no arguments it means to install prebuilt scripts, if you pass, `hex|path|git`, it means installing from a project (which therefore requires building and mix).

So have we settled on the following syntax?

```
# this PR
escript.install X    # X is a path or a URL to a prebuilt escript file

# future PR
escript.install X Y [Z]  
  # X is the package type (hex|path|git),
  # Y is the package name or URL
  # Z would be a version for a Hex package, or a ref for a Git repo
```

In that case, we can drop the `./` prefix when installing a file from the local directory. It was used to disambiguate from the `mix escript.install <hex-package>` syntax.